File: 75 days Challenge Leetcode/Array+String/Day-05/Solution.cpp
#include <string>
using namespace std;

class Solution {
public:
    // Helper function to check if a character is a vowel
    bool isVowel(char c) {
        return c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u' ||
               c == 'A' || c == 'E' || c == 'I' || c == 'O' || c == 'U';
    }
    
    // Function to reverse only the vowels in a string
    string reverseVowels(string s) {
        int i = 0;  // Pointer starting from the beginning of the string
        int j = s.length() - 1;  // Pointer starting from the end of the string
        
        // While the two pointers have not crossed each other
        while (i < j) {
            // Move the left pointer until we find a vowel
            while (i < j && !isVowel(s[i])) {
                i++;
            }
            // Move the right pointer until we find a vowel
            while (i < j && !isVowel(s[j])) {
                j--;
            }
            
            // If we have found vowels at both pointers, swap them
            if (i < j) {
                swap(s[i], s[j]);
                i++;  // Move the left pointer forward
                j--;  // Move the right pointer backward
            }
        }
        
        return s;  // Return the modified string with vowels reversed
    }
};

File: 75 days Challenge Leetcode/Array+String/Day-05/Solution.py
class Solution:
    # Helper function to check if a character is a vowel
    def isVowel(self, c: str) -> bool:
        return c in 'aeiouAEIOU'
    
    # Function to reverse only the vowels in a string
    def reverseVowels(self, s: str) -> str:
        s = list(s)  # Convert the string to a list of characters
        i, j = 0, len(s) - 1  # Two pointers starting from the beginning and end
        
        # While the two pointers have not crossed each other
        while i < j:
            # Move the left pointer until we find a vowel
            while i < j and not self.isVowel(s[i]):
                i += 1
            # Move the right pointer until we find a vowel
            while i < j and not self.isVowel(s[j]):
                j -= 1
            
            # If we have found vowels at both pointers, swap them
            if i < j:
                s[i], s[j] = s[j], s[i]
                i += 1  # Move the left pointer forward
                j -= 1  # Move the right pointer backward
        
        return ''.join(s)  # Convert the list back to a string
